common/src/confidential/crypto.rs

```diff
@@ -108,9 +108,9 @@ fn encode_encryption(
 /// Returns a tuple of each component.
 fn split_encrypted_payload(
     data: Vec<u8>,
-) -> Fallible<(PublicKey, u64, u64, Vec<u8>, Vec<u8>, Nonce)> {
+) -> Result<(PublicKey, u64, u64, Vec<u8>, Vec<u8>, Nonce)> {
     if data.len() < PublicKey::len() + NONCE_SIZE + CIPHER_LEN_SIZE + AAD_LEN_SIZE {
-        return Err(format_err!("invalid nonce or public key"));
+        return Err(anyhow!("invalid nonce or public key"));
```

Comment:
you can replace the whole thing with [ensure!](https://docs.rs/anyhow/1.0.31/anyhow/macro.ensure.html)
```rust
ensure!(data.len() >= PublicKey::len() + NONCE_SIZE + CIPHER_LEN_SIZE + AAD_LEN_SIZE,
	"invalid nonce or public key")